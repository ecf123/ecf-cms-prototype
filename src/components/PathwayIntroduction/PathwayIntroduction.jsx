import "./PathwayIntroduction.scss";
import React from "react";

const PathwayIntroduction = (props) => {
  const { image, description } = props;
  return (
    <div className="pathway-introduction">
      <img
        className="pathway-introduction__image"
        src={image}
        alt="pathway thumbnail"
      />
      <h2 className="pathway-introduction__title">Pathway Overview</h2>
      <p className="pathway-introduction__description">{description}</p>
    </div>
  );
};

export default PathwayIntroduction;
