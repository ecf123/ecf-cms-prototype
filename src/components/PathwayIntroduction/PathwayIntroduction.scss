@use "../../styles/variables" as *;

.pathway-introduction {
  display: grid;
  grid-template-columns: min-content auto;
  grid-template-rows: auto 1fr;
  gap: 1rem;
  padding: 2rem;
  max-width: 56rem;

  &__image {
    grid-area: 1 / 1 / 3 / 2;
  }

  &__title {
    color: $color-blue-navy-dark;
    font-size: $font-size-medium;
  }

  &__description {
    color: $color-teal;
    font-size: $font-size-extra-small;
  }
}
