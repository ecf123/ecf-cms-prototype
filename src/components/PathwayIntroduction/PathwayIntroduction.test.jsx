import PathwayIntroduction from "./PathwayIntroduction";
import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";


describe("PathwayIntroduction component", () => {
    it("matches the snapshot taken", () => {
        const { container } = customRender(<PathwayIntroduction />);
        expect(container).toMatchSnapshot();
    });

    it("renders the image", () => {
        render(<PathwayIntroduction/>);
        const image = screen.getByAltText("pathway thumbnail");
        expect(image).toBeInTheDocument();
    })

    it("renders the description", () => {
        render(<PathwayIntroduction description={"Description"}/>);
        const description = screen.getByText("Description");
        expect(description).toBeInTheDocument();
    })

})