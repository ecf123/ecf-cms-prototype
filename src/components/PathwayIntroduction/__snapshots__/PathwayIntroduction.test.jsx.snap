// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PathwayIntroduction component matches the snapshot taken 1`] = `
<div>
  <div
    class="pathway-introduction"
  >
    <img
      alt="pathway thumbnail"
      class="pathway-introduction__image"
    />
    <h2
      class="pathway-introduction__title"
    >
      Pathway Overview
    </h2>
    <p
      class="pathway-introduction__description"
    />
  </div>
</div>
`;
